handshake/stk_generator.go

```diff
@@ -0,0 +1,72 @@
+package handshake
+
+import (
+	"bytes"
+	"net"
+	"time"
+
+	"github.com/lucas-clemente/quic-go/crypto"
+)
+
+// An STK is a source address token
+type STK struct {
+	RemoteAddr string
+	SentTime   time.Time
+}
+
+// An STKGenerator generates STKs
+type STKGenerator struct {
```

Comment:
That's not what I meant. I think the stuff in the crypto package should just treat the token as a byte slice that is boxed / unboxed. The timestamp can then be (de)serialized in the handshake packet. This is how the our crypto/handshake package split usually works.

For simplicity we could then even use Go's binary encoding of the struct, so we don't even need parsing logic ourselves (which is always a win).